### PR TITLE
#1374: Proposed API -- DO NOT MERGE

### PR DESCRIPTION
**Pull request by njbartlett**
Proposed API for template post-processors

### CODE REVIEW

File: org.bndtools.templating/src/org/bndtools/templating/PostProcessor.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public interface PostProcessor {
+
+    /**
+     * <p>
+     * Service registration property indicating the kind(s) of templates this post-processor can target. Currently
+     * defined values are 'project', 'workspace' and 'bndrun'. The type of this property is String+. The post-processor
+     * shall be ignored if it does not specify at least one kind.
+     * </p>
+     */
+    public static String PROP_TEMPLATE_KIND = "kind";
```
Comment (fhuberts) [inline org.bndtools.templating/src/org/bndtools/templating/PostProcessor.java]: why not use an array?
why not use an enum?

File: org.bndtools.templating/src/org/bndtools/templating/PostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     *            The input parameters that were supplied to the template.
+     * @param resources
+     *            The resources generated by the template. A post-processor may add, remove or modify any entries in
+     *            this map.
+     * @param monitor
+     *            A monitor for progress reporting.
+     * @return
+     * @throws Exception
+     */
+    IStatus modifyOutputs(String templateKind, Template template, Map<String,List<Object>> templateParams, ResourceMap resources, IProgressMonitor monitor) throws Exception;
```
Comment (fhuberts) [inline org.bndtools.templating/src/org/bndtools/templating/PostProcessor.java]: If at all possible I'd like this decoupled of Eclipse interfaces.
bndtools could provide internal conversion.

I gather that a plugin can modify the template and when I return bndtools will actually generate them on the filesystem?

**Comment by bjhargrave** [comment]
@njbartlett What is going on with this? Should this be closed or is there progress to be made?

**Comment by njbartlett** [comment]
Don’t close yet please, I need to pick this up again soon.

**Pull request closed without merging**